Arrow/Sources/Arrow/ArrowArrayBuilder.swift

```diff
@@ -169,11 +169,36 @@ public class StructArrayBuilder: ArrowArrayBuilder<StructBufferBuilder, StructAr
         let arrowData = try ArrowData(self.type, buffers: buffers,
                                       children: childData, nullCount: self.nullCount,
                                       length: self.length)
-        let structArray = try StructArray(arrowData)
+        let structArray = try NestedArray(arrowData)
         return structArray
     }
 }
 
+public class ListArrayBuilder: ArrowArrayBuilder<ListBufferBuilder, NestedArray> {
+    let valueBuilder: any ArrowArrayHolderBuilder
+
+    public override init(_ elementType: ArrowType) throws {
```

Comment:
This doesn't allow element nullability to be specified. I think this isn't actually enforced (which is a separate issue) but the public API should reflect the spec at least. I'd do something like this, directly providing the list type which holds the `ArrowField` and therefore the nullability information;

```
    public override init(_ arrowType: ArrowType) throws {
        guard let arrowType = arrowType as? ArrowTypeList else {
            throw ArrowError.invalid("Expected ArrowTypeList")
        }
        let arrowField = arrowType.elementField
        self.valueBuilder = try ArrowArrayBuilders.loadBuilder(arrowType: arrowField.type)
        try super.init(arrowType)
    }
```